In [3]:
import pandas as pd

Kim Rexell Laviña: Activity 1: Briefly discuss your respective dataset. Provide the source of your dataset too.

Kenneth Popera: Activity 2: Perform loading / reading your group’s dataset

Loading the dataset requires the pandas library which is used for reading, writing and manipulating data. In order to load the csv file of our dataset read_csv() method is used to read the contents of the csv file.

In [4]:
dataset = pd.read_csv('breast_cancer_wisconsin_csv.csv', index_col = False, names = ['id', 'clump_thickness', 'cell_size_uniformity', 'cell_shape_uniformity', 'marginal_adhesion', 'single_epithelial_cell_size', 'bare_nuclei', 'bland_chromatin', 'normal_nucleoli', 'mitoses', 'class'])
dataset

id  clump_thickness  cell_size_uniformity  cell_shape_uniformity  \
0    1000025                5                     1                      1   
1    1002945                5                     4                      4   
2    1015425                3                     1                      1   
3    1016277                6                     8                      8   
4    1017023                4                     1                      1   
..       ...              ...                   ...                    ...   
694   776715                3                     1                      1   
695   841769                2                     1                      1   
696   888820                5                    10                     10   
697   897471                4                     8                      6   
698   897471                4                     8                      8   

     marginal_adhesion  single_epithelial_cell_size bare_nuclei  \
0                    1                            2           1   
1                    5                            7          10   
2                    1                            2           2   
3                    1                            3           4   
4                    3                            2           1   
..                 ...                          ...         ...   
694                  1                            3           2   
695                  1                            2           1   
696                  3                            7           3   
697                  4                            3           4   
698                  5                            4           5   

     bland_chromatin  normal_nucleoli  mitoses  class  
0                  3                1        1      2  
1                  3                2        1      2  
2                  3                1        1      2  
3                  3                7        1      2  
4                  3                1        1      2  
..               ...              ...      ...    ...  
694                1                1        1      2  
695                1                1        1      2  
696                8               10        2      4  
697               10                6        1      4  
698               10                4        1      4  

[699 rows x 11 columns]

Marben Vilaflor: Activity 3:  Identify the shape of your data and discuss

Getting the shape of our dataset will use the numpy library. First, you need to convert the dataset into numpy array and print the shape of the dataset using shape() method of numpy library. There are 699 rows and 11 columns in our dataset.

In [5]:
import numpy as np

data_shape = np.array(dataset)
print(np.shape(data_shape))

(699, 11)


Kyle Joseph Timajo: Activity 4: Enlist the column names in your dataset and provide the type of data

The Wisconsin Breast Cancer Dataset column names are:    
   1. id number
   2. Clump Thickness               1 - 10
   3. Uniformity of Cell Size       1 - 10
   4. Uniformity of Cell Shape      1 - 10
   5. Marginal Adhesion             1 - 10
   6. Single Epithelial Cell Size   1 - 10
   7. Bare Nuclei                   1 - 10
   8. Bland Chromatin               1 - 10
   9. Normal Nucleoli               1 - 10
  10. Mitoses                       1 - 10
  11. Class:                        (2 for benign, 4 for malignant)

All of the columns except id number and class has range of values from 1 to 10. The class has a value of either 2 (benign) or 4 (malignant). The id number composed of random numbers. To display all the columns we used the keys() method from pandas. Then, for the data types is we used dtypes.


In [11]:
columns = dataset.keys()
print(columns)

data_types = dataset.dtypes
print(data_types)

Index(['id', 'clump_thickness', 'cell_size_uniformity',
       'cell_shape_uniformity', 'marginal_adhesion',
       'single_epithelial_cell_size', 'bare_nuclei', 'bland_chromatin',
       'normal_nucleoli', 'mitoses', 'class'],
      dtype='object')
id                              int64
clump_thickness                 int64
cell_size_uniformity            int64
cell_shape_uniformity           int64
marginal_adhesion               int64
single_epithelial_cell_size     int64
bare_nuclei                    object
bland_chromatin                 int64
normal_nucleoli                 int64
mitoses                         int64
class                           int64
dtype: object


Kim Rexell Laviña: Activity 5: Provide a sample view of your dataset

In [13]:
dataset

id  clump_thickness  cell_size_uniformity  cell_shape_uniformity  \
0    1000025                5                     1                      1   
1    1002945                5                     4                      4   
2    1015425                3                     1                      1   
3    1016277                6                     8                      8   
4    1017023                4                     1                      1   
..       ...              ...                   ...                    ...   
694   776715                3                     1                      1   
695   841769                2                     1                      1   
696   888820                5                    10                     10   
697   897471                4                     8                      6   
698   897471                4                     8                      8   

     marginal_adhesion  single_epithelial_cell_size bare_nuclei  \
0                    1                            2           1   
1                    5                            7          10   
2                    1                            2           2   
3                    1                            3           4   
4                    3                            2           1   
..                 ...                          ...         ...   
694                  1                            3           2   
695                  1                            2           1   
696                  3                            7           3   
697                  4                            3           4   
698                  5                            4           5   

     bland_chromatin  normal_nucleoli  mitoses  class  
0                  3                1        1      2  
1                  3                2        1      2  
2                  3                1        1      2  
3                  3                7        1      2  
4                  3                1        1      2  
..               ...              ...      ...    ...  
694                1                1        1      2  
695                1                1        1      2  
696                8               10        2      4  
697               10                6        1      4  
698               10                4        1      4  

[699 rows x 11 columns]

Kenneth Popera: Activity 6: Provide a sample view of some dirty data in your dataset. i.e missing values or unusual values in your dataset. You may count the number of rows that contains missing values.

As provided in the dataset there are 16 missing values and it is denoted by question mark "?". All of the missing values are in the bare_nuclei column.

In [14]:
dataset.head(42)

id  clump_thickness  cell_size_uniformity  cell_shape_uniformity  \
0   1000025                5                     1                      1   
1   1002945                5                     4                      4   
2   1015425                3                     1                      1   
3   1016277                6                     8                      8   
4   1017023                4                     1                      1   
5   1017122                8                    10                     10   
6   1018099                1                     1                      1   
7   1018561                2                     1                      2   
8   1033078                2                     1                      1   
9   1033078                4                     2                      1   
10  1035283                1                     1                      1   
11  1036172                2                     1                      1   
12  1041801                5                     3                      3   
13  1043999                1                     1                      1   
14  1044572                8                     7                      5   
15  1047630                7                     4                      6   
16  1048672                4                     1                      1   
17  1049815                4                     1                      1   
18  1050670               10                     7                      7   
19  1050718                6                     1                      1   
20  1054590                7                     3                      2   
21  1054593               10                     5                      5   
22  1056784                3                     1                      1   
23  1057013                8                     4                      5   
24  1059552                1                     1                      1   
25  1065726                5                     2                      3   
26  1066373                3                     2                      1   
27  1066979                5                     1                      1   
28  1067444                2                     1                      1   
29  1070935                1                     1                      3   
30  1070935                3                     1                      1   
31  1071760                2                     1                      1   
32  1072179               10                     7                      7   
33  1074610                2                     1                      1   
34  1075123                3                     1                      2   
35  1079304                2                     1                      1   
36  1080185               10                    10                     10   
37  1081791                6                     2                      1   
38  1084584                5                     4                      4   
39  1091262                2                     5                      3   
40  1096800                6                     6                      6   
41  1099510               10                     4                      3   

    marginal_adhesion  single_epithelial_cell_size bare_nuclei  \
0                   1                            2           1   
1                   5                            7          10   
2                   1                            2           2   
3                   1                            3           4   
4                   3                            2           1   
5                   8                            7          10   
6                   1                            2          10   
7                   1                            2           1   
8                   1                            2           1   
9                   1                

Marben Villaflor: Activity 7: Perform data cleaning and provide a screenshot of both the dirty data (in 5.) and a clean data.

In cleaning the dataset, we remove all the rows that contains "?". From 699 rows, it became 683 rows after cleaning. To remove all the rows with question marks, we loop through all the rows and check each value of the the bare_nuclei column and drop the row using the drop() method of pandas.

In [15]:
for x in dataset.index:
  if dataset.loc[x, "bare_nuclei"] == "?":
    dataset.drop(x, inplace = True)

In [19]:
dataset

id  clump_thickness  cell_size_uniformity  cell_shape_uniformity  \
0    1000025                5                     1                      1   
1    1002945                5                     4                      4   
2    1015425                3                     1                      1   
3    1016277                6                     8                      8   
4    1017023                4                     1                      1   
..       ...              ...                   ...                    ...   
694   776715                3                     1                      1   
695   841769                2                     1                      1   
696   888820                5                    10                     10   
697   897471                4                     8                      6   
698   897471                4                     8                      8   

     marginal_adhesion  single_epithelial_cell_size bare_nuclei  \
0                    1                            2           1   
1                    5                            7          10   
2                    1                            2           2   
3                    1                            3           4   
4                    3                            2           1   
..                 ...                          ...         ...   
694                  1                            3           2   
695                  1                            2           1   
696                  3                            7           3   
697                  4                            3           4   
698                  5                            4           5   

     bland_chromatin  normal_nucleoli  mitoses  class  
0                  3                1        1      2  
1                  3                2        1      2  
2                  3                1        1      2  
3                  3                7        1      2  
4                  3                1        1      2  
..               ...              ...      ...    ...  
694                1                1        1      2  
695                1                1        1      2  
696                8               10        2      4  
697               10                6        1      4  
698               10                4        1      4  

[683 rows x 11 columns]

Kyle Joseph Timajo: Activity 8: Compute the mean, media, mode, standard deviation, variance, ranges and percentiles of your dataset. Identify possible outliers.

